UnityProject/Assets/Scripts/Health/RespiratoryHealth/RespiratorySystem.cs

```diff
@@ -268,6 +370,177 @@ private void HandleBreathingPlasma(float plasmaAmount)
+			{
+				return true;
+			}
+
+			// 10% chances of message
+			var theirPronoun = gameObject.Player() != null
+				? gameObject.Player().Script.characterSettings.TheirPronoun()
+				: "its";
+			Chat.AddActionMsgToChat(
+				gameObject,
+				CO2LowYouMessages.PickRandom(),
+				string.Format(
+					CO2LowOthersMessages.PickRandom(),
+					gameObject.ExpensiveName(),
+					string.Format(plasmaLowOthersMessages.PickRandom(), gameObject.ExpensiveName(), theirPronoun))
+			);
+		}
+		// enough carbon dioxide to start suffocating you!
+		else if (carbonPressure > CARBON_DIOXIDE_SAFE_MAX)
+		{
+			if (CarbonSuffocationTimer >= 0)
+				CarbonSuffocationTimer++;
+			if (CarbonSuffocationTimer >= 3 && CarbonSuffocationTimer < 7)
+			{
+				IsSuffocating = true;
+				bloodSystem.OxygenDamage += 3;
+
+				if (DMMath.Prob(90))
+				{
+					return true;
+				}
+
+				// 10% chances of message
+				var theirPronoun = gameObject.Player() != null
+					? gameObject.Player().Script.characterSettings.TheirPronoun()
+					: "its";
+				Chat.AddActionMsgToChat(
+					gameObject,
+					CO2MedYouMessages.PickRandom(),
+					string.Format(CO2MedOthersMessages.PickRandom(), gameObject.ExpensiveName(), theirPronoun)
+				);
+			}
+			else if (CarbonSuffocationTimer >= 7)
+			{
+				IsSuffocating = true;
+				bloodSystem.OxygenDamage += 8;
+
+				if (DMMath.Prob(90))
+				{
+					return true;
+				}
+
+				// 10% chances of message
+				var theirPronoun = gameObject.Player() != null
+					? gameObject.Player().Script.characterSettings.TheirPronoun()
+					: "its";
+				Chat.AddActionMsgToChat(
+					gameObject,
+					CO2HighYouMessages.PickRandom(),
+					string.Format(CO2HighOthersMessages.PickRandom(), gameObject.ExpensiveName(), theirPronoun)
+				);
+				return true;
+			}
+		}
+
+		if (carbonPressure < CARBON_DIOXIDE_WARNING_LEVEL)
+		{
+			CarbonSuffocationTimer = 0;
+			return (carbonPressure > 0);
+		}
+		return false;
+	}
+	private bool HandleWearingGasMask(GasMix gasMix)
+	{
+
+		// if there is too much CO2 in the air
+		if (gasMix.GetMoles(Gas.CarbonDioxide) >= 30)
+		{
+			// if there is too much CO2 in the air, but there isn't too much plasma in the air
+			if (gasMix.GetMoles(Gas.Plasma) < 25)
+			{
+				GasMix gasMix2 = gasMix;
+				gasMix2.RemoveGas(Gas.CarbonDioxide, 30);
+				HandleBreathingCarbonDioxide(gasMix2);
+				return true;
+			}
+			else if (gasMix.GetMoles(Gas.Plasma) >= 25)
+			{
+				GasMix gasMix2 = gasMix;
+				gasMix2.RemoveGas(Gas.Plasma, 25);
+				gasMix2.RemoveGas(Gas.CarbonDioxide, 30);
+				float plasmaBreathedWithMask = HandleBreathingPlasma(gasMix2);
+				if (plasmaBreathedWithMask > 0)
+				{
+					gasMix.RemoveGas(Gas.Plasma, plasmaBreathedWithMask);
+					registerTile.Matrix.MetaDataLayer.UpdateSystemsAt(registerTile.LocalPositionClient, SystemType.AtmosSystem);
+				}
+				HandleBreathingCarbonDioxide(gasMix2);
+				return true;
+			}
+		}
+		// if there is too much plasma in the air
+		if (gasMix.GetMoles(Gas.Plasma) >= 25)
```

Comment:
Won't both this block and the one above it run if there's too much plasma and co2 in the air? Why do the check on the inside in the first place? Can't you just check plasma, do the plasma things, then check co2 and do the co2 things?

Reply:
i believe you missed the "return" present in the block above. but you do make a good point.